march_hardware_interface/src/march_hardware_interface.cpp

```diff
@@ -173,25 +164,19 @@ void MarchHardwareInterface::init()
     MarchTemperatureSensorHandle marchTemperatureSensorHandle(joint_names_[i], &joint_temperature_[i],
                                                               &joint_temperature_variance_[i]);
     march_temperature_interface.registerHandle(marchTemperatureSensorHandle);
+  }
 
-    // Enable high voltage on the IMC
-    if (joint.canActuate())
-    {
-      if (hasPowerDistributionBoard)
-      {
-        int net_number = joint.getNetNumber();
-        if (net_number != -1)
-        {
-          marchRobot.getPowerDistributionBoard()->getHighVoltage().setNetOnOff(true, net_number);
-        }
-        else
-        {
-          ROS_FATAL("Joint %s has no high voltage net number", joint.getName().c_str());
-          throw std::runtime_error("Joint has no high voltage net number");
-        }
-      }
-      joint.prepareActuation();
-    }
+  this->prepareJointsForActuation();
+  bool encoderSetCorrectly = this->IMotionCubesWorkingCorrectly();
+
+  while (!encoderSetCorrectly)
```

Comment:
```suggestion
  while (!this->IMotionCubesWorkingCorrectly())
```
an extra boolean is not really needed. You can just use the method directly, right?